superbench/benchmarks/micro_benchmarks/tensorrt_inference_performance.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""TensorRT inference micro-benchmark."""
+
+import re
+from pathlib import Path
+
+import torch.onnx
+import torchvision.models
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, Platform, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class TensorRTInferenceBenchmark(MicroBenchmarkWithInvoke):
+    """TensorRT inference micro-benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'trtexec'
+        self._pytorch_models = [
+            'resnet50',
+            'resnet101',
+            'resnet152',
+            'densenet169',
+            'densenet201',
+            'vgg11',
+            'vgg13',
+            'vgg16',
+            'vgg19',
+        ]
+        self.__model_cache_path = Path.home() / '.cache/torch/hub/checkpoints'
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--pytorch_models',
+            type=str,
+            nargs='+',
+            default=self._pytorch_models,
+            help='ONNX models for TensorRT inference benchmark, e.g., {}.'.format(', '.join(self._pytorch_models)),
+        )
+
+        self._parser.add_argument(
+            '--batch_size',
+            type=int,
+            default=32,
+            required=False,
+            help='Set batch size for implicit batch engines.',
+        )
+
+        self._parser.add_argument(
+            '--iterations',
+            type=int,
+            default=256,
+            required=False,
+            help='Run at least N inference iterations.',
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        self.__bin_path = str(Path(self._args.bin_dir) / self._bin_name)
+
+        for model in self._args.pytorch_models:
+            if hasattr(torchvision.models, model):
+                torch.onnx.export(
+                    getattr(torchvision.models, model)(pretrained=True).cuda(),
+                    torch.randn(self._args.batch_size, 3, 224, 224, device='cuda'),
+                    f'{self.__model_cache_path / (model + ".onnx")}',
+                )
+                self._commands.append(
+                    ' '.join(
+                        [
+                            self.__bin_path,
+                            '--int8',
```

Comment:
It should be a configuration? Not hard code

Reply:
any other option for inference? is it necessary to run fp16?

Reply:
updated